src/annotator/test/guest-test.js

```diff
@@ -420,15 +420,18 @@ describe('Guest', () => {
     });
 
     describe('on "deleteAnnotation" event', () => {
-      it('detaches annotation', () => {
+      it('prevents saving of annotations that are currently anchoring', async () => {
         const guest = createGuest();
-        const callback = sinon.stub();
-        guest._emitter.subscribe('anchorsChanged', callback);
+        const ann = { target: [], uri: 'uri', $tag: 'tag1' };
 
-        emitSidebarEvent('deleteAnnotation', 'tag1');
+        emitSidebarEvent('loadAnnotations', [ann]);
+        emitSidebarEvent('deleteAnnotation', ann.$tag);
```

Comment:
Deleting this line test passes, so this is not ideal.

Reply:
Why not add a target here?